### PR TITLE
Story: sync passages

### PR DESCRIPTION
**Pull request by youbitchoc**
host decides

### CODE REVIEW

**Comment by youbitchoc** [comment]
are there any worldloading race conditions i should look out for?

File: Story/StoryHooks.cs [hunk 1]
```diff
@@ -802,6 +809,95 @@ private void RainWorldGame_GoToDeathScreen(On.RainWorldGame.orig_GoToDeathScreen
...
+            orig(self, manager, ID);
+            if (isStoryMode(out _) && !OnlineManager.lobby.isOwner)
+            {
+                StoryMenuHelpers.RemoveMenuObjects(self.mapButtonPrompt, self.prevButton, self.nextButton, self.chooseButton, self.buttonInstruction);
+            }
+        }
+
+        private void FastTravelScreen_ctor_ClientDontFilterRegions(ILContext il)
+        {
+            try
```
Comment (6fears7) [inline Story/StoryHooks.cs:842]: Nitpick: commenting the changes to the IL. I can infer from the method name though. Your call.

**Review by 6fears7 — COMMENTED** [review]
I realize this got converted back to a draft but I'll just post the comment for now.

File: Story/StoryHooks.cs (hunk 1 above)
Comment (youbitchoc) [reply Story/StoryHooks.cs:842]: it's just
```diff
  bool flag = false;
+ if (isStoryMode(out _) && !OnlineManager.lobby.isOwner) flag = true;
```

and to understand what flag does you'll have to look at the IL anyway

**Comment by henpemaz** [comment]
> are there any worldloading race conditions i should look out for?

The passage-screen performs world-loading to get room data and map data but it might already have been addressed that it doesn't require/requests the resource.

Then finally using a passage also performs a quick world-loading where it loads the game in the now-modified save position, saves and immediately exits out into a new rainworldgame instance to re-start the cycle. This is iirc detectable with the manager.menusettings.newgamecondition.IsFastTravel, see where and how that's used by the game.

Imo, this "passage handling" cycle-load is probably best to have be performed only in the host and have clients be waiting in menus until the real-deal cycle starts.

**Comment by youbitchoc** [comment]
> The passage-screen performs world-loading to get room data and map data but it might already have been addressed that it doesn't require/requests the resource.

yup, fast travel screen uses a null game so it gets ignored

> Imo, this "passage handling" cycle-load is probably best to have be performed only in the host

aight

File: Story/StoryHooks.cs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 using HUD;
+using IL.Menu;
```
Comment (henpemaz) [inline Story/StoryHooks.cs:2]: this looks like it could cause trouble

**Comment by henpemaz** [comment]
other than that seems alright at a glance, I didn't check the specifics of the new hooks

File: Story/StoryHooks.cs (hunk 2 above)
Comment (youbitchoc) [reply Story/StoryHooks.cs]: whoops, fixed

**Comment by youbitchoc** [comment]
the passage ones are for stopping client from changing region and syncing host selected region

other significant change is using default denpos from synced savestring rather than syncing separately (now it's just used as an indicator for passage / custom ending denpos, not happy with it but it works well enough)

**Review by 6fears7 — APPROVED** [review]

**Pull request merged**